backend/routes/RoutesCommon.go

```diff
@@ -0,0 +1,96 @@
+package routes
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"database/sql"
+
+	"github.com/ucsdscheduleplanner/UCSD-Schedule-Planner/backend/db"
+)
+
+/*
+queryAndResponse is a function to
+query using ds
+process rows using rowScanner
+write response
+*/
+func queryAndResponse(ds *db.DatabaseStruct, tag string, writer http.ResponseWriter, request *http.Request,
+	rowScanner func(*sql.Rows) (val interface{}, err error),
```

Comment:
Possibly improve this by receiving a struct "[Scanner](https://golang.org/pkg/database/sql/#Scanner)" or have some default rowScanner

Reply:
Don't know if this works

```golang
func rowScannerInterface() {
	var val interface{}
	err := rows.Scan(&val)
	return val, err
}
```